src/tokens.ts

```diff
@@ -249,19 +250,32 @@ export class TokenComputer {
 
     extractIdentifierFromExtendedIdentifier(identifier: string): string {
         const parts = identifier.split("-");
+        const { prefix, ticker, randomSequence } = this.splitIdentifierIntoComponents(parts);
 
         this.checkIfExtendedIdentifierWasProvided(parts);
-        this.ensureTokenTickerValidity(parts[0]);
-        this.checkLengthOfRandomSequence(parts[1]);
+        this.ensureTokenTickerValidity(ticker);
+        this.checkLengthOfRandomSequence(randomSequence);
+        if (prefix) {
+            this.checkLengthOfPrefix(prefix);
+            return prefix + "-" + ticker + "-" + randomSequence;
+        }
+        return ticker + "-" + randomSequence;
+    }
+
+    private splitIdentifierIntoComponents(identifier: string): { prefix: any; ticker: any; randomSequence: any } {
+        const parts = identifier.split("-");
+        if (parts.length >= 3 && parts[2].length === this.TOKEN_RANDOM_SEQUENCE_LENGTH) {
```

Comment:
This condition is not fully correct. `NFT-1q2w3e-100000` will be considered with prefix and is not correct.
Another way maybe is to check:
- if parts[0] is full lowercase alphanumeric => starts with prefix
- else if parts[0] is full uppercase alphanumeric => starts with ticker
- otherwise throw error